### PR TITLE
STYLE: fix pylint simplifiable-if-statement warnings

### PR DESCRIPTION
**Pull request by Moisan**
Related to https://github.com/pandas-dev/pandas/issues/48855
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/dtypes/dtypes.py [hunk 1]
```diff
@@ -1254,10 +1251,7 @@ def is_dtype(cls, dtype: object) -> bool:
         if isinstance(dtype, str):
             if dtype.lower().startswith("interval"):
                 try:
-                    if cls.construct_from_string(dtype) is not None:
-                        return True
-                    else:
-                        return False
+                    return bool(cls.construct_from_string(dtype) is not None)
```
Comment (mzeitlin11) [inline pandas/core/dtypes/dtypes.py:1254]: `is` should always give boolean so could lose the `bool`? Same above

File: pandas/io/formats/info.py [hunk 2]
```diff
@@ -579,10 +576,7 @@ def memory_usage_bytes(self) -> int:
         memory_usage_bytes : int
             Object's total memory usage in bytes.
         """
-        if self.memory_usage == "deep":
-            deep = True
-        else:
-            deep = False
+        deep = bool(self.memory_usage == "deep")
```
Comment (mzeitlin11) [inline pandas/io/formats/info.py:579]: Similar comment - could you remove the `bool` here?

File: pandas/tests/tseries/offsets/test_week.py [hunk 3]
```diff
@@ -114,11 +114,7 @@ def test_is_on_offset(self, weekday):
 
         for day in range(1, 8):
             date = datetime(2008, 1, day)
-
-            if day % 7 == weekday:
-                expected = True
-            else:
-                expected = False
+            expected = bool(day % 7 == weekday)
```
Comment (mzeitlin11) [inline pandas/tests/tseries/offsets/test_week.py:117]: Same for this one also

**Review by mzeitlin11 — COMMENTED** [review]
Thanks for the pr @Moisan! Small comment on further cleanup

**Review by MarcoGorelli — APPROVED** [review]
Thanks @Moisan

**Pull request merged**